rllib/execution/parallel_requests.py

```diff
@@ -173,12 +179,13 @@ def get_ready(self) -> Dict[ActorHandle, List[Any]]:
             objs = ray.get(ready_requests)
         else:
             objs = ready_requests
+
         for req, obj in zip(ready_requests, objs):
             actor = self._pending_to_actor[req]
             self._remote_requests_in_flight[actor].remove(req)
             ready_requests_dict[actor].append(obj)
             del self._pending_to_actor[req]
-        del ready_requests
```

Comment:
this may be added because of a weird mem leak ...

Reply:
Oh boy. @avnishn , are we sure this line avoids a real leak? In general, wouldn't leaving this line mean that we also would have to `del` everywhere else? I mean literally *everywhere* ? :)